transaction/extra/src/tx_summary_unblinding/verifier.rs

```diff
@@ -143,7 +155,7 @@ impl TxSummaryStreamingVerifier {
     /// Returns:
     /// * A properly initialized TxSummaryStreamingVerifier
     pub fn new(
-        extended_message_digest: &[u8; 32],
+        extended_message_digest: &[u8],
```

Comment:
Is there a good reason for this change?

I think this may change the meaning of the hash at line 154/166 in this diff. When you digest a fixed size array, I don't remember if it prepends the length. Even if it works, what advantage is there to this?

Reply:
It seems much better to fail fast int he extended message digest is not 32 bytes -- this will always be broken and give a garbage output if it is not 32 bytes.

Reply:
the MCIP for tx summary verification is only applicable when the block version is relatively late, at a place where extended message digest is 32 bytes

Reply:
> It seems much better to fail fast if the extended message digest is not 32 bytes -- this will always be broken and give a garbage output if it is not 32 bytes.

the `heapless::Vec<u8; 32>` i'm using for storing messages doesn't implement `&[u8; 32]`... but i can probably swap this with an `Option<[u8; 32]>` if there is _no case_ where a message / digest will ever be a different length?

Reply:
can we do like `heapless::Vec<u8; 32>::as_ref().try_into()?` or something like this, at the call site?

i feel like it's okay to make the caller responsible for checking this